cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -362,6 +372,14 @@ boolean scanApplications(ScanApplicationsRequest request, int txBatchSize,
    */
   Collection<ApplicationSpecification> getAllAppVersions(ApplicationId id);
 
+  /**
+   * Returns the latest version of an application
+   *
+   * @param id application id
+   * @return The metadata information of the latest application version.
+   */
+  ApplicationMeta getLatestAppVersion(ApplicationId id);
+
```

Comment:
Need this function to fetch the latest version of the app, used in multiple scenarios:
In the deploy action - a check to compare current deployment request with the latest, 
In the `GET /apps/<app-id>` to retrieve the latest version information, 
and all other apis mentioned here: [go/pipeline-lcm-rest-endpoints](http://go/pipeline-lcm-rest-endpoints)

Reply:
Just call it `getLatest`.

Reply:
this is kind of a weird API because ApplicationId already contains a version. Would be more accurate to take a NamespaceId and application name

Reply:
Yes changed it to namespace and appId